sa/storage-authority.go

```diff
@@ -58,190 +176,155 @@ func NewSQLStorageAuthority(driver string, name string) (ssa *SQLStorageAuthorit
 }
 
 func (ssa *SQLStorageAuthority) InitTables() (err error) {
-	tx, err := ssa.db.Begin()
+	ssa.dbMap.AddTableWithName(Registration{}, "registrations").SetKeys(false, "ID").SetVersionCol("LockCol")
+	ssa.dbMap.AddTableWithName(Pending_auth{}, "pending_authz").SetKeys(false, "ID").SetVersionCol("LockCol")
+	ssa.dbMap.AddTableWithName(Auth{}, "authz").SetKeys(false, "ID")
+	ssa.dbMap.AddTableWithName(Certificate{}, "certificates").SetKeys(false, "Serial")
+	ssa.dbMap.AddTableWithName(CertificateStatus{}, "certificateStatus").SetKeys(false, "Serial").SetVersionCol("LockCol")
+	ssa.dbMap.AddTableWithName(OcspResponse{}, "ocspResponses").SetKeys(true, "ID")
+	ssa.dbMap.AddTableWithName(Crl{}, "crls").SetKeys(false, "Serial")
+
+	err = ssa.dbMap.CreateTablesIfNotExists()
+	return
+}
+
+func (ssa *SQLStorageAuthority) DumpTables() {
+	fmt.Printf("===== TABLE DUMP =====\n")
+
+	fmt.Printf("\n----- registrations -----\n")
+	var registrations []Registration
+	_, err := ssa.dbMap.Select(&registrations, "SELECT * FROM registrations ")
 	if err != nil {
+		fmt.Println(err)
 		return
 	}
+	for _, r := range registrations {
+		fmt.Printf("%+v\n", r)
+	}
 
-	// All fields should be created with "NOT NULL" because go's SQL support does not
-	// handle null values well (see, e.g. https://github.com/go-sql-driver/mysql/issues/59)
-	statements := []string{
-
-		// Create registrations table
-		// TODO: Add NOT NULL to thumbprint and value.
-		`CREATE TABLE IF NOT EXISTS registrations (
-		id VARCHAR(255) NOT NULL,
-		thumbprint VARCHAR(255),
-		value BLOB
-	);`,
-
-		// Create pending authorizations table
-		// TODO: Add NOT NULL to value. Right now it causes test failures because some
-		// inserts to not fill all fields.
-		`CREATE TABLE IF NOT EXISTS pending_authz (
-		id VARCHAR(255) NOT NULL,
-		value BLOB
-	);`,
-
-		// Create finalized authorizations table
-		`CREATE TABLE IF NOT EXISTS authz (
-		sequence INTEGER NOT NULL,
-		id VARCHAR(255) NOT NULL,
-		digest TEXT NOT NULL,
-		value BLOB NOT NULL
-	);`,
-
-		// Create certificates table. This should be effectively append-only, enforced
-		// by DB permissions.
-		`CREATE TABLE IF NOT EXISTS certificates (
-		serial VARCHAR(255) PRIMARY KEY NOT NULL,
-		digest VARCHAR(255) NOT NULL,
-		value BLOB NOT NULL,
-		issued DATETIME NOT NULL
-		);`,
-
-		// Create certificate status table. This provides metadata about a certificate
-		// that can change over its lifetime, and rows are updateable unlike the
-		// certificates table. The serial number primary key matches up with the one
-		// on certificates.
-		// subscriberApproved: 1 iff the subscriber has posted back to the server
```

Comment:
The comments in this file have value, we should move them to the corresponding model. Alternately, for ones where the mode closely matches the struct in objects.go, move the comments to the canonical struct in objects.go (or skip, if the comments in objects.go are already sufficient).